src/common/meta/src/key.rs

```diff
@@ -590,6 +593,69 @@ impl TableMetadataManager {
         Ok(())
     }
 
+    /// Restores metadata for table.
+    /// The caller MUST ensure it has the exclusive access to `TableNameKey`.
+    pub async fn restore_table_metadata(
```

Comment:
I think it's better if we bring the "move" operation back, instead of setting the values again. WDYT @fengjiachun

Reply:
I don't want the `move` back to `KvBackend`; I want to keep it simple.

Reply:
I prefer to use `move_table_metadata`, and then completely delete it in the last step of the procedure; this way, if something goes wrong, the data can be fully recovered. The cost is an extra RPC call, but the process is actually more elegant because values are not fetched up, they are moved directly. This means that the upper layer does not touch the data and cannot cause any data contamination.